### ISSUE TITLE
Scaler are removed twice while shutting down service

### ISSUE THREAD

**Issue by vtdat** [vCloud-DFTBA/faythe#38]
Steps to reproduce:
* Create cloud provider
* Create scaler
* CTRL + C or SIGINT 

Expected:
* Service exited

Actual:
* Scaler are removed twice leading to panic

```log
level=info ts=2019-11-12T03:27:07.428Z caller=main.go:186 msg="Received SIGTERM, exiting gracefully..."
level=info ts=2019-11-12T03:27:07.428Z caller=manager.go:68 component="autoscale manager" msg="Stopping autoscale manager..."
level=info ts=2019-11-12T03:27:07.429Z caller=manager.go:112 component="autoscale manager" msg="Removing scaler" id=/scalers/9b7835fa8189f79c9e4d3b26a51aa5dc/b4904bdcd2c9675a8e432898d0311bb8
level=info ts=2019-11-12T03:27:07.429Z caller=manager.go:112 component="autoscale manager" msg="Removing scaler" id=/scalers/9b7835fa8189f79c9e4d3b26a51aa5dc/b4904bdcd2c9675a8e432898d0311bb8
level=debug ts=2019-11-12T03:27:07.429Z caller=scaler.go:67 component="autoscale manager" scaler=/scalers/9b7835fa8189f79c9e4d3b26a51aa5dc/b4904bdcd2c9675a8e432898d0311bb8 msg="Scaler is stopping" id=b4904bdcd2c9675a8e432898d0311bb8
level=debug ts=2019-11-12T03:27:07.429Z caller=scaler.go:67 component="autoscale manager" scaler=/scalers/9b7835fa8189f79c9e4d3b26a51aa5dc/b4904bdcd2c9675a8e432898d0311bb8 msg="Scaler is stopping" id=b4904bdcd2c9675a8e432898d0311bb8
level=debug ts=2019-11-12T03:27:07.429Z caller=scaler.go:70 component="autoscale manager" scaler=/scalers/9b7835fa8189f79c9e4d3b26a51aa5dc/b4904bdcd2c9675a8e432898d0311bb8 msg="Scaler is stopped" id=b4904bdcd2c9675a8e432898d0311bb8
panic: close of closed channel
```

**Comment by ntk148v** [comment]
Relate to #40

**Comment by vtdat** [comment]
Fixes in #40
